jcommune-model/src/main/java/org/jtalks/jcommune/model/entity/Topic.java

```diff
@@ -766,4 +769,32 @@ public int getUserPostCount(JCUser user) {
         }
         return count;
     }
+
+    /**
+     * Makes URL to mark topic page as read.
+     * For anonymous user returns empty optional.
+     *
+     * @param user current user
+     * @param page page to mark as read
+     * @return Optional url string
+     */
+    public Optional<String> getMarkAsReadUrl(JCUser user, String page) {
+        if (user.isAnonymous()) {
+            return Optional.absent();
+        }
+        return Optional.of(URL_TEMPLATE.replace(TOPIC_ID, String.valueOf(getId()))
+                                       .replace(PAGE_NUM, page)
+                                       .replace(USER_ID, String.valueOf(user.getId()))
+                                       .replace(LAST_MODIFIED, String.valueOf(getLastModificationPostDate().getMillis()))
+        );
+    }
+
+    static class MarkAsReadUrlConstants {
+        static final String URL_TEMPLATE = "{topicId}/page/{pageNum}/markread?userId={userId}&lastModified={lastModified}";
+
+        static final String TOPIC_ID = "{topicId}";
```

Comment:
Can we just hardcode these in the method? They don't seem to be used anywhere else.